### PR TITLE
Handle conditional access operator in CompareSymbolsCorrectlyFix

### LINKED ISSUE #4337: Use symbols correctly: code fix bug with null-conditional access operator
Caught this while applying the code fix to:

https://github.com/dotnet/roslyn-analyzers/blob/91b109f2f69496b76e7a740a4547a9d9b9a12c0b/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L296

It was transformed into:

```csharp
if (self? SymbolEqualityComparer.Default.Equals(self, overload) == true)
```

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #4337

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs [hunk 1]
```diff
@@ -580,6 +575,66 @@ void Method1({symbolType} x , {symbolType} y)
...
+}
+";
+
+            await new VerifyCS.Test
+            {
+                TestState = { Sources = { source, SymbolEqualityComparerStubCSharp } },
+                FixedState = { Sources = { fixedSource, SymbolEqualityComparerStubCSharp } },
+            }.RunAsync();
+        }
+
```
Comment (Evangelink) [inline src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs:637]: Please add a test (c# and vbnet) with a chain call like `a?.b?.c?.Equals(d)`.

**Review by Evangelink — APPROVED** [review]
LGTM - add the requested tests.

File: src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs [hunk 2]
```diff
@@ -580,6 +575,166 @@ void Method1({symbolType} x , {symbolType} y)
...
+            await new VerifyVB.Test
+            {
+                TestState = { Sources = { source, SymbolEqualityComparerStubVisualBasic } },
+                FixedState = { Sources = { fixedSource, SymbolEqualityComparerStubVisualBasic } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task CompareSymbolFromInstanceEqualsWithChainNullConditionalAccess_CSharp()
```
Comment (Youssef1313) [inline src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs:685]: This is not currently passing. I'll give it a look when I get sometime (the new college semester just started, so might not be very soon). Feel free if anyone would like to give it a look before I get to it.

**Review by mavasani — APPROVED** [review]

**Comment by Evangelink** [comment]
@mavasani I was able to make some good progresses on this PR, I just need your input on how can I recreate a `ConditionalExpressionSyntax` (it is internal on the version of the generator we use, seems like it is public on newer versions).

**Comment by Evangelink** [comment]
> @mavasani I was able to make some good progresses on this PR, I just need your input on how can I recreate a `ConditionalExpressionSyntax` (it is internal on the version of the generator we use, seems like it is public on newer versions).

ping @mavasani

**Comment by mavasani** [comment]
@Evangelink You can probably copy over the Roslyn C# and VB syntax generator implementations directly into the fixer's C# and VB implementations.

**Pull request closed without merging**